slack_ros/scripts/slack_ros.py

```diff
@@ -34,46 +34,72 @@ def __init__(self):
         self.username = rospy.get_param('~username', 'ros-bot')
 	
         # Create a publisher for our custom message.
-        pub = rospy.Publisher('from_slack_to_ros', String, queue_size=10)
-	rospy.Subscriber("from_ros_to_slack", String, self.callback)
-	rospy.Subscriber("send_file_to_slack", String, self.filecallback)
+        self.pub = rospy.Publisher('from_slack_to_ros', String, queue_size=10)
+        rospy.Subscriber("from_ros_to_slack", String, self.message_callback)
+        rospy.Subscriber("send_file_to_slack", String, self.file_callback)
 
-	# Create the slack client
-	self.sc = SlackClient(self.token)
+        # Create the slack client
+        self.slack_client = SlackClient(self.token)
+        self.has_connection = False
+        self.subscribed_msg_store = []
```

Comment:
there should be no need to save messages in a queue ourselves. ROS subscriber (with queue size) will handle that for us. we just have to block the callback method until we've reconnected to the slack service. ROS will not execute callbacks in parallel but sequentially as soon as the subscriber callback finished. Thus this is exactly the behaviour we're looking for.

Please use a configurable queue size and remove ```subscribed_msg_store```